### PR TITLE
Cleanup of plotting.plot_implicit

### PR DESCRIPTION
**Pull request by oscargus**
#### Brief description of what is fixed or changed
Adds explicit arguments to plotting.plot_implicit according to #14030 and corrected/improved the documentation

#### Release Notes

* plotting
  * Added explicit optional arguments to plot_implicit and improved/corrected the documentation

### CODE REVIEW

**Comment by oscargus** [comment]
I didn't fully get the testing initially but relied on studying the diffs. However, a tab sneaked in (darn auto-indenting on empty lines...). I made a fixup and force push on the update, but clearly the tests didn't restart. Is it possible to do that manually or should I just make a dummy commit to retrigger it? Also holds for #15757 (From now on I will run the quality check locally...)

**Comment by oscargus** [comment]
Is there a way to run the single failing test locally? I do not fully understand where and why it fails...

File: sympy/plotting/plot_implicit.py [hunk 1]
```diff
@@ -200,7 +200,8 @@ def _get_meshes_grid(self):
 
 
 @doctest_depends_on(modules=('matplotlib',))
-def plot_implicit(expr, x_var=None, y_var=None, **kwargs):
+def plot_implicit(expr, x_var=None, y_var=None, adaptive=True, depth=0,
+                  points=300, line_color="Blue", show=True, **kwargs):
```
Comment (asmeurer) [inline sympy/plotting/plot_implicit.py]: I don't know if it matters, but `blue` was changed from lowercase to uppercase here.

Comment (oscargus) [reply sympy/plotting/plot_implicit.py]: Better not risk anything. I'll change it.

**Pull request merged**